client/basic-authorship/src/basic_authorship.rs

```diff
@@ -910,7 +915,9 @@ mod tests {
 
 		let proposer = block_on(proposer_factory.init(&genesis_header)).unwrap();
 
-		// Give it enough time
+		// Exact block_limit, which includes:
+		// 99 (header_size) + 718 (proof@initialize_block) + 246 (one Transfer extrinsic)
+		let block_limit = 1063;
```

Comment:
I just don't know if we can in advance compute the proof size value, which would be there during the block construction ([here](https://github.com/paritytech/substrate/blob/14f1a3951d5a9de4dd3ae3ff83fd5ef66c4fbbe7/client/basic-authorship/src/basic_authorship.rs#L428-L451)). The size of the block is also _estimated_ there (as the method name states).

The previous test was also silently assuming that proof size will just be smaller then some arbitrarily choosen value (size of 4 extrinsics) which was not really related to proof size.

Maybe the proof size could be captured right after just calling `runtiem_api.initialize_block`, this could work, but I was not exploring that.

For me hardcoded value seems to be acceptable here, if you think it is abuse I can put some more work into this. wdyt?

Reply:
I digged a bit. One option could be to locally instantiate a temporary `block_builder` just to get a "portable" solution.

```
let builder = client.new_block_at(genesis_header.hash(), Default::default(), true);
let block_limit = builder.estimate_block_size(true) + extrinsics[0].encoded_size();
```

This gives indeed 1063.

If you think is an overkill just ignore me. Your choice